compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/SymbolFinder.java

```diff
@@ -1265,9 +1265,16 @@ public void visit(BLangClassDefinition classDefinition) {
         lookupNodes(classDefinition.referencedFields);
         lookupNode(classDefinition.initFunction);
         lookupNodes(classDefinition.functions);
+        lookupAnnAttachmentsAttachedToMethods(classDefinition.functions);
         lookupNodes(classDefinition.typeRefs);
     }
 
+    private void lookupAnnAttachmentsAttachedToMethods(List<BLangFunction> functions) {
```

Comment:
What if we add a `lookupNodes(List<BLangFunction> functions)` instead (which overloads the `lookupNodes()` method)? @KavinduZoysa @dulajdilshan

Reply:
If we use `lookupNodes(List<BLangFunction> functions)`, it is looking up each function, but what we need to do here is, look up function annotations separately. Therefore we need a separate method like this.

Reply:
Yes, inside `lookupNodes(List<BLangFunction>)` we can lookup only the attachments:
```
for (BLangFunction f : functions) {
    lookupNodes(f.annotationAttachments);
    lookupNode(f);
}
```
WDYT?

Reply:
Yes this logic should work, but I think we cannot have a function like 
```ballerina
    private void lookupNodes(List<BLangFunction> functions) {

    }
```
since it matches the generic function. We may need to change the function name

Reply:
Got it. No problem then